### ISSUE TITLE
Support for compound joins in SQL context

### ISSUE THREAD

**Issue by aaarrti** [pola-rs/polars#11290]
### Problem
Hi, 
do you plan to support compound joins for SQL context? E.g., 

```python
import polars as pl

df1 = pl.DataFrame({
    "a": [1, 2, 3, 4, 5],
    "b": [1, 3, 4, 4, 5]
})

df2 = pl.DataFrame({
    "a": [1, 2, 3, 4, 5],
    "b": [0, 3, 4, 5, 6]
})

# This one works like a charm
df1.join(df2, on=["a", "b"])

# And this one fails
ctx = pl.SQLContext(df1=df1, df2=df2)
ctx.execute("SELECT * FROM df1 JOIN df2 ON df1.a = df2.a AND df1.b = df2.b", eager=True)

```
The error shown is 
```
InvalidOperationError: SQL join constraint On(BinaryOp { left: BinaryOp { left: CompoundIdentifier([Ident { value: "df1", quote_style: None }, Ident { value: "a", quote_style: None }]), op: Eq, right: CompoundIdentifier([Ident { value: "df2", quote_style: None }, Ident { value: "a", quote_style: None }]) }, op: And, right: BinaryOp { left: CompoundIdentifier([Ident { value: "df1", quote_style: None }, Ident { value: "b", quote_style: None }]), op: Eq, right: CompoundIdentifier([Ident { value: "df2", quote_style: None }, Ident { value: "b", quote_style: None }]) } }) is not yet supported
```

Based on the error, and the fact, that the same join works using dataframes directly, I assume the logic is already implemented in the query engine, but the respective parsing logic is missing from the `SQLContext`. Did I understand this correctly? 

### Questions
- Do you have plans to extend `SQLContext` to support compound joins?
- How high would be the development effort to add this feature?
- How high would the impact on the existing codebase be? In case it is minor/localised, I'd be happy to contribute myself.

Thanks in advance!

**Comment by Icequeen3333** [comment]
I think the exception is formed with the `ON` operator, as using the `USING` operator does not result in an error.

`ctx = pl.SQLContext(df1=df1, df2=df2).execute("SELECT * FROM df1 JOIN df2 USING(a,b)", eager=True)`

Although this is still an issue, compound joins are supported, not with the `ON` operator.

**Comment by alexander-beedie** [comment]
> * How high would be the development effort to add this feature?

After fixing the issue with `USING` I know exactly where this patch would need to go, and the answer appears to be "not very much"  ;)

> * How high would the impact on the existing codebase be? In case it is minor/localised, I'd be happy to contribute myself.

Marginal - your contribution would be very welcome! Happy to point you in the right direction if you want to have a go at this one? :)

**Comment by aaarrti** [comment]
Hi @alexander-beedie thanks! I'd love to give it a try, so far, your explanations have already pointed me in the (right) direction 🙃

**Comment by alexander-beedie** [comment]
> Edit: Although using doesn't generate an error, it also doesn't work with more than one parameter, which is probably also a bug. Assigning this the bug label would also be accurate I believe.

@Icequeen3333: FYI, we just released `0.19.8` which has a fix for this `USING` issue (not respecting multiple cols) 👌

**Comment by alexander-beedie** [comment]
Closed by https://github.com/pola-rs/polars/pull/13242 - thanks @aaarrti 👍
